bin/setup

```diff
@@ -56,7 +56,6 @@ Dir.chdir APP_ROOT do
   run "bundle check || bundle install --without deploy production"
   run "yarn install"
   run "gem install thin -v 1.5.1 -- --with-cflags=\"-Wno-error=implicit-function-declaration\""
-  run "gem install mailcatcher -- --with-cppflags=-I/usr/local/opt/openssl/include"
```

Comment:
I feel like we're always running into issues with new folks installing things with the right flags, is there a way to keep these flag instructions in the Gemfile? [slack link](https://gsa-tts.slack.com/archives/C0NGESUN5/p1657058930249809)

Reply:
I have two questions about this:
1. Is it required that (mac) users install openssl via brew? I assume this is because of the lack of header files in the default install if so
2. Homebrew has changed its directory structure for M1 macs (learned this yesterday). Perhaps we can just case on the system information and provide the correct flag in each case?

Reply:
+1 for moving to Gemfile. I did hit dependency errors though with all recent versions of `mailcatcher` :/. (Both Eric and I use M1s)

Reply:
I think this stems from the author recommending against installing Mailcatcher via bundler: https://mailcatcher.me/#bundler

Conflicting versions strikes me as a potential problem, but I'd be fine with it as long as there are no conflicts.

Reply:
@pauldoomgov, @mitchellhenke mentioned you might have a M1 too? Have you run into this?

Reply:
@mitchellhenke you're right, I forgot about that. I introduced that problem, even though we still have the original problem.

Reply:
an observation: the versions of `mailcatcher` and `thin` that this installs are from 2011 and 2013, respectively. I think I'm going to try and get current versions working and keep it out of bundler.

Reply:
Do we even need `thin` ? 🤔

Reply:
I think mailcatcher needs thin... which gives me another idea 💡 what if we had `Gemfile.mailcatcher` and updated our scripts to use that completely separately?

```
BUNDLE_GEMFILE=Gemfile.mailcatcher bundle exec mailcatcher
```

```
# Gemfile.mailcatcher

gem 'mailcatcher'
```